### PR TITLE
Add segment replication cluster-level setting

### LINKED ISSUE #4197: [DOC][Segment Replication] Add Documentation for new cluster setting to set default index replication type
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [ X] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.

 Add Documentation for new cluster setting to set default index replication type. By using this new cluster setting, users will be able to set default replication type for their newly created indices. This way users can enable segment replication on all new indices by using the cluster setting.

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
[7420](https://github.com/opensearch-project/OpenSearch/pull/7420)

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Fixes #4197 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kolchfa-aws** [comment]
@Rishikesh1159 Could you please review this PR for technical accuracy? Thanks!

**Review by Rishikesh1159 — APPROVED** [review]

**Review by anasalkouz — APPROVED** [review]

**Review by cwillum — APPROVED** [review]
LGTM

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md [hunk 1]
```diff
@@ -30,9 +30,13 @@ Segment replication is the first feature in a series of features designed to dec
 
 ## Segment replication configuration
 
-To set segment replication as the replication strategy, create an index with replication.type set to `SEGMENT`:
+Setting the default replication type on a cluster affects all newly created indexes. However, you can specify a different replication type when creating an index. Index-level settings always override cluster-level settings.
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:33]:
```suggestion
Setting the default replication type for a cluster affects all newly created indexes. However, you can specify a different replication type when creating an index. Index-level settings always override cluster-level settings.
```

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md [hunk 2]
```diff
@@ -30,9 +30,13 @@ Segment replication is the first feature in a series of features designed to dec
...
 
-````json
+### Creating an index with segment replication type
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:35]: "Setting segment replication as the replication type"?

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md [hunk 3]
```diff
@@ -65,6 +69,37 @@ curl -X PUT "$host/_cluster/settings?pretty" -H 'Content-Type: application/json'
 ```
 {% include copy-curl.html %}
 
+### Setting the replication type on a cluster
+
+You can set the default replication type for newly created indexes in a cluster in the `opensearch.yml` file as follows:
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:74]:
```suggestion
You can set the default replication type for newly created cluster indexes in the `opensearch.yml` file as follows:
```

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md [hunk 4]
```diff
@@ -65,6 +69,37 @@ curl -X PUT "$host/_cluster/settings?pretty" -H 'Content-Type: application/json'
...
+
+```yaml
+cluster.indices.replication.strategy: 'SEGMENT'
+```
+{% include copy.html %}
+
+This cluster-level setting cannot be enabled through the REST API.
+{: .note}
+
+This setting is not applied for system indexes and hidden indexes. By default, all system and hidden indexes in OpenSearch will still use document replication even if this setting is enabled.
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:84]:
```suggestion
This setting is not applied to system indexes and hidden indexes. By default, all system and hidden indexes in OpenSearch will still use document replication even if this setting is enabled.
```

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md (hunk 2 above)
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:35]:
```suggestion
### Creating an index with the segment replication type
```

File: _tuning-your-cluster/availability-and-recovery/segment-replication/index.md [hunk 5]
```diff
@@ -65,6 +69,37 @@ curl -X PUT "$host/_cluster/settings?pretty" -H 'Content-Type: application/json'
...
+{: .note}
+
+### Creating an index with document replication type
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:87]: "Setting document replication as the replication type"?

Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/segment-replication/index.md:87]:
```suggestion
### Creating an index with the document replication type
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!

**Pull request merged**